internal/internal_workflow_client.go

```diff
@@ -1708,45 +1747,90 @@ func (w *workflowClientInterceptor) UpdateWorkflow(
 			},
 		},
 	})
-	handle := &updateHandle{ref: resp.GetUpdateRef()}
 	if err != nil {
-		handle.err = err
-		return handle, nil
+		return nil, err
```

Comment:
This change is a compatibility break and it actually breaks the update-enablement-test in the features repo but I think we should do it...

There are two ways to deal with the sync/async wait policies: one way is to spawn a goroutine to do the rpc and return the handle immediately with a future in it (in all cases). The second way is to not return the handle until the RPC returns and there's no background goro to manage.

In the first case, all errors have to be tunneled through the Handle.Get call, even early server errors like a security error (which is what happens when update is disabled on the server). In the second approach where the initial UpdateWorkflowExecution waits for the RPC to return, if the server returns an early error then we have the opportunity to return the error directly and never a Handle. 

I want to use the second approach and that is what is implemented here.  In addition to being simpler in concept and implementation it also sidesteps an awkward situation where the RunID is not necessarily known until the RPC returns so what should the handle do in that case? A second Future for the RunID? Tell clients spin on GetRunID()?

Reply:
Ok - did this. Renamed as suggested and turned the interface into a 3-field struct to be passed by value.